src/include/OpenImageIO/fmath.h

```diff
@@ -1538,29 +1683,44 @@ inline float fast_sinh (float x) {
         r = madd(r * a, a2, a);
         return copysignf(r, x);
     }
+#else
+    return sinhf(x);
+#endif
 }
 
-inline float fast_cosh (float x) {
+inline OIIO_HOSTDEVICE float fast_cosh (float x) {
+#ifndef __CUDACC__
     // Examined 2237485550 values of cosh on [-87.3300018,87.3300018]: 1.78256726 avg ulp diff, 178 max ulp
     float e = fast_exp(fabsf(x));
     return 0.5f * e + 0.5f / e;
+#else
+    return coshf(x);
+#endif
 }
 
-inline float fast_tanh (float x) {
+inline OIIO_HOSTDEVICE float fast_tanh (float x) {
+#ifndef __CUDACC__
     // Examined 4278190080 values of tanh on [-3.40282347e+38,3.40282347e+38]: 3.12924e-06 max error
     // NOTE: ulp error is high because of sub-optimal handling around the origin
     float e = fast_exp(2.0f * fabsf(x));
     return copysignf(1 - 2 / (1 + e), x);
+#else
+    return tanhf(x);
+#endif
 }
 
-inline float fast_safe_pow (float x, float y) {
+inline OIIO_HOSTDEVICE float fast_safe_pow (float x, float y) {
     if (y == 0) return 1.0f; // x^0=1
     if (x == 0) return 0.0f; // 0^y=0
     // be cheap & exact for special case of squaring and identity
     if (y == 1.0f)
         return x;
     if (y == 2.0f)
+#ifndef __CUDACC__
         return std::min (x*x, std::numeric_limits<float>::max());
+#else
+        return min (x*x, std::numeric_limits<float>::max());
```

Comment:
Why does this use `min`, but below in interpolate_linear, you use `fminf`?

Reply:
Poor temporal locality when editing? I'll make them consistent.